### PR TITLE
Add Closeable/AutoCloseable on Database & interface

### PR DESCRIPTION
**Pull request by zorglube**
## Impact

- [ ] Bug fix (non-breaking change which fixes expected existing functionality)
- [x] Enhancement/New feature (adds functionality without impacting existing logic)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
 
## Description

Added the interface `AutoCloseable` to the interfaces `DataBase` and `DatabaseConnection` , in order to provid the possibility to use Java `try with resources`.

### CODE REVIEW

**Comment by zorglube** [comment]
This PR is related to this Issue https://github.com/liquibase/liquibase/issues/2974

**Comment by zorglube** [comment]
@kataggart I need a first review please.

**Comment by zorglube** [comment]
> @kataggart I need a first review please.

Please.

**Comment by nvoxland** [comment]
It's looking good so far. Both already have close() methods on them, so it makes sense and won't impact the APIs.

You added the try with resources to the tests, you can add `throws Exception` to the test method signature vs. just logging the error like you have.

Besides that, I have a vague hope to someday remove the need for a "close()" interface from Database in general to make that class more about being a "dialect" vs. another layer of a connection. But that is already going to be a large enough change that it would be nice to have Autoclosable on Database for now.

**Comment by zorglube** [comment]
> You added the try with resources to the tests, you can add `throws Exception` to the test method signature vs. just logging the error like you have.

I'll work on that, then ping you back.

**Comment by zorglube** [comment]
@nvoxland : I updated the unit tests. IF it's good for you, do you have any idea of when the code will be merge ?

**Review by nvoxland — APPROVED** [review]

**Comment by nvoxland** [comment]
It looks good, @zorglube . I made it a non-draft release and kicked off a build on it to make sure everything is still all passing. After those finish we have a couple other internal review & test steps but I'd imagine it will be in the next release. We generally shoot for a new release every couple weeks and just had one recently, so I'd guess it will be around then.

**Comment by nvoxland** [comment]
Code review and test results:

Things to be aware of:
- Only "shipped" change is adding the Closable interface to Database and DatabaseConnection. Both of which already defined the close() method so no real code impact beyond that extra marker interface.
- The rest of the changes are fixes to tests to avoid linters complaining about "not closing autoclosable objects"

Things to worry about:
- Nothing

**Review by suryaaki2 — APPROVED** [review]

File: liquibase-core/src/main/java/liquibase/database/core/DatabaseUtils.java [hunk 1]
```diff
@@ -39,11 +41,22 @@ public static void initializeDatabase(String defaultCatalogName, String defaultS
                 String searchPath = executor.queryForObject(new RawSqlStatement("SHOW SEARCH_PATH"), String.class);
 
                 if (!searchPath.equals(defaultCatalogName) && !searchPath.startsWith(defaultSchemaName + ",") && !searchPath.startsWith("\"" + defaultSchemaName + "\",")) {
-                    if (database instanceof CockroachDatabase) {
```
[the remark this answers is not in the document]
Comment (nvoxland) [reply liquibase-core/src/main/java/liquibase/database/core/DatabaseUtils.java]: A sync from master made it show a lot of commits that aren't actually new for this PR. I changed the PR to a point to a different branch and then back to master which cleans it up. It's a github UI thing.

**Comment by zorglube** [comment]
Thanks guys! ☺️

**Pull request merged**